shopfloor_reception_mobile/static/src/scenario/reception.js

```diff
@@ -0,0 +1,723 @@
+            const next_unhandled_line = this.state.data.selected_move_lines.find(
+                (line) => line.qty_done < line.quantity
+            );
+            return this._get_selected_line_ids([next_unhandled_line]);
+        },
+        _apply_search_filter: function (picking, input) {
+            const values = [picking.origin];
+            return !_.isEmpty(values.find((v) => v.includes(input)));
+        },
+        _get_selected_line_ids: function (lines) {
+            return lines.map(_.property("id"));
+        },
+    },
+    data: function () {
+        return {
+            usage: "reception",
+            initial_state_key: "select_document",
+            scan_destination_qty: 0,
+            states: {
+                init: {
+                    enter: () => {
+                        this.wait_call(this.odoo.call("start"));
+                    },
+                },
+                select_document: {
+                    display_info: {
+                        title: "Choose an operation",
+                        scan_placeholder: "Scan document / product / package",
+                    },
+                    events: {
+                        select: "on_select",
+                    },
+                    on_select: (selected) => {
+                        this.wait_call(
+                            this.odoo.call("scan_document", {
+                                barcode: selected.name,
+                            })
+                        );
+                    },
+                    on_scan: (barcode) => {
+                        this.wait_call(
+                            this.odoo.call("scan_document", {
+                                barcode: barcode.text,
+                            })
+                        );
+                    },
+                },
+                select_line: {
+                    display_info: {
+                        title: "Select a line",
+                        scan_placeholder: "Scan product / package",
+                    },
+                    events: {
+                        mark_as_done: "on_mark_as_done",
+                        cancel_picking_line: "on_cancel",
+                    },
+                    on_scan: (barcode) => {
+                        this.wait_call(
+                            this.odoo.call("scan_line", {
+                                picking_id: this.state.data.picking.id,
+                                barcode: barcode.text,
+                            })
+                        );
+                    },
+                    on_mark_as_done: () => {
+                        this.wait_call(
+                            this.odoo.call("done_action", {
+                                picking_id: this.state.data.picking.id,
+                            })
+                        );
+                    },
+                    on_cancel: () => {
+                        // TODO: endpoint missing in backend
```

Comment:
??

Reply:
It was decided with @jbaudoux that we'd leave this implementation in the backend for a second version of the scenario.
I left this in the frontend so that it's easier to implement later.